src/pages/App.js

```diff
@@ -1,16 +1,72 @@
-import React from 'react'
-import 'bootstrap/dist/css/bootstrap.css'
+/* eslint-disable no-nested-ternary */
+/* eslint-disable no-unused-expressions */
+import React, { useEffect } from 'react'
 import { Container, Row } from 'react-bootstrap'
+import 'bootstrap/dist/css/bootstrap.css'
+import { connect, useDispatch } from 'react-redux'
 // eslint-disable-next-line import/extensions
 import Header from '../features/Header/Header.jsx'
 // eslint-disable-next-line import/extensions
 import MoviesSearch from '../features/MoviesSearch'
 // eslint-disable-next-line import/extensions
 import MoviesCards from '../features/MoviesCards'
-// eslint-disable-next-line import/extensions
 import Footer from '../features/Footer/Footer.jsx'
+import { useGetMoviesQuery } from '../services/api.js'
+import {
+  clearSearchMovies,
+  setMovies,
+  totalAmount,
+} from '../redux/actions/actions.js'
+
+/*
+  =============================================================
+
+  This is the second option
+
+  const xhttp = new XMLHttpRequest()
+  xhttp.onreadystatechange = function () {
+    if (xhttp.readyState === 4 && xhttp.status === 200) {
+      console.log(xhttp.responseText)
+    }
+  }
+
+  xhttp.open('GET', 'http://localhost:4000/movies?limit=10', true)
+  xhttp.send()
+
+  =============================================================
+*/
+
+function App({
+  settingMovies,
+  settingTotalAmount,
+  sortBy,
+  sortOrder,
+  filter,
+  limit,
+  search,
+  setClearSearchMovies,
+}) {
+  const { data, isSuccess, isError, error, isLoading } = useGetMoviesQuery({
```

Comment:
With Redux all side-effects should be done inside of the Redux. That's why we need saga or thunk libs: to make async actions possible in Redux

Everything you did here with data should be moved in some Redux action like getMoviesData (result of this action will be = movies data is fetched, settingMovies or settingTotalAmount should not be exposed to React at all - it shouldn't know about existance of something where data is set per se), in app you'd be dispatching it and all requesting and setting data in the store will happen under the Redux hood. React doesn't orchestrate how data is handled, written or read. It just calls simplest action possible

Reply:
So you want me to change to redux thunk | saga from rtk query completely. Wow, there is so many things that I must change. implementing, testing... 😮😐

Reply:
![image](https://user-images.githubusercontent.com/71421741/181186028-2cc9f82a-34b9-4f49-83b0-fb2720c181f3.png)